test/apps/iodemo/io_demo.cc

```diff
@@ -752,22 +819,24 @@ class DemoClient : public P2pDemoCommon {
             MemoryPool<IoReadResponseCallback>& pool) :
             _comp_counter(0), _io_counter(NULL), _server_io_counter(NULL),
             _sn(0), _validate(false), _iov(NULL), _buffer(malloc(buffer_size)),
-            _buffer_size(buffer_size), _pool(pool) {
+            _buffer_size(buffer_size), _meta_comp_counter(0), _pool(pool) {
 
             if (_buffer == NULL) {
                 throw std::bad_alloc();
             }
         }
 
         void init(long *io_counter, long *conn_io_counter,
-                  uint32_t sn, bool validate, BufferIov *iov) {
+                  uint32_t sn, bool validate, BufferIov *iov,
+                  int meta_comp_counter = 1) {
             /* wait for all data chunks and the read response completion */
-            _comp_counter      = iov->size() + 1;
+            _comp_counter      = iov->size() + meta_comp_counter;
             _io_counter        = io_counter;
             _server_io_counter = conn_io_counter;
             _sn                = sn;
             _validate          = validate;
             _iov               = iov;
+            _meta_comp_counter = meta_comp_counter;
```

Comment:
do we really need separate counter?

Reply:
> it helps to distinguish tag and AM cases.
maybe add something clearer if it's needed?

> In the latter we do not need to do IOMSG validation
why?

Reply:
added comment

Reply:
ok, i don't have other comments